core/src/main/java/com/expediagroup/streamplatform/streamregistry/core/services/StreamService.java

```diff
@@ -93,7 +102,17 @@
 
   @PreAuthorize("hasPermission(#stream, 'DELETE')")
   public void delete(Stream stream) {
-    throw new UnsupportedOperationException();
+    handlerService.handleDelete(stream);
```

Comment:
Sorry, I keep coming back to this function...

With the move from Service to Repo, it looks like we have introduced the ability to leave orphaned `[Consumer|Producer]Bindings` again. I could be missing something.

Personally, I'm not sure on the feasibility, but I'd love to see an IT that proves that deleting a stream also removes all referenced entities.

Reply:
added util service for consumer/producer binding delete to be used when deleting stream or stream binding